plugins/kubernetes/app/models/kubernetes/deploy_executor.rb

```diff
@@ -142,22 +142,36 @@ def fetch_pods
       end
     end
 
+    def show_logs_on_deploy_if_requested(statuses)
+      pods = statuses.map(&:pod).compact.select { |p| p.annotations[:'samson/show_logs_on_deploy'] == 'true' }
+      pods.each { |pod| print_pod_details(pod, Time.now, events: false) }
```

Comment:
I think `Time.now` is calculated on each iteration, and as retrieving the pods can take some seconds (if some timeout happens, for example), the logs are not for the same ""snapshot"" on different pods. I don't think this is a problem and this way is simple and nice, IMHO.

But just something worth noting, perhaps.